encoding/kzg/prover/v2/prover.go

```diff
@@ -434,8 +291,72 @@ func (p *Prover) createIcicleBackendProver(
 	return CreateIcicleBackendProver(p, params, fs)
 }
 
+func (g *Prover) preloadAllEncoders() error {
+	paramsAll, err := getAllPrecomputedSrsMap(g.KzgConfig.CacheDir)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("detect %v srs maps\n", len(paramsAll))
+	for i := 0; i < len(paramsAll); i++ {
+		fmt.Printf(" %v. NumChunks: %v   ChunkLength: %v\n", i, paramsAll[i].NumChunks, paramsAll[i].ChunkLength)
+	}
+
+	if len(paramsAll) == 0 {
+		return nil
+	}
+
+	for _, params := range paramsAll {
+		// get those encoders and store them
+		enc, err := g.GetKzgEncoder(params)
+		if err != nil {
+			return err
+		}
+		g.ParametrizedProvers[params] = enc
+	}
+
+	return nil
+}
+
+// Detect the precomputed table from the specified directory
+// the file name follow the name convention of
+//
+//	dimE*.coset&
+//
+// where the first * specifies the dimension of the matrix which
+// equals to the number of chunks
+// where the second & specifies the length of each chunk
+func getAllPrecomputedSrsMap(tableDir string) ([]encoding.EncodingParams, error) {
+	files, err := os.ReadDir(tableDir)
+	if err != nil {
+		return nil, fmt.Errorf("read srs table dir: %w", err)
+	}
+
+	tables := make([]encoding.EncodingParams, 0)
+	for _, file := range files {
+		filename := file.Name()
+
+		tokens := strings.Split(filename, ".")
+
+		dimEValue, err := strconv.Atoi(tokens[0][4:])
+		if err != nil {
+			return nil, fmt.Errorf("parse dimension part of the table: %w", err)
+		}
+		cosetSizeValue, err := strconv.Atoi(tokens[1][5:])
+		if err != nil {
+			return nil, fmt.Errorf("parse coset size part of the table: %w", err)
+		}
+
+		params := encoding.EncodingParams{
+			NumChunks:   uint64(cosetSizeValue),
```

Comment:
Actually I noticed today that this is wrong. NumChunks should be dimE and chunkLength should be cosetSize, even the comment above says that. I don't think this code path has ever been used...